microservices-connector/cmd/router/main.go

```diff
@@ -589,29 +983,53 @@ func mcGraphHandler(w http.ResponseWriter, req *http.Request) {
 
 			// Write the chunk to the ResponseWriter
 			if _, err := w.Write(buffer[:n]); err != nil {
-				log.Error(err, "failed to write to ResponseWriter")
+				otlpr.WithContext(log, ctx).Error(err, "failed to write to ResponseWriter")
+				spanTokens.RecordError(err)
+				spanTokens.SetStatus(codes.Error, err.Error())
+				spanTokens.End()
 				return
 			}
+
 			// Flush the data to the client immediately
 			if flusher, ok := w.(http.Flusher); ok {
 				flusher.Flush()
 			} else {
-				log.Error(errors.New("unable to flush data"), "ResponseWriter does not support flushing")
+				err := errors.New("unable to flush data")
+				otlpr.WithContext(log, ctx).Error(err, "ResponseWriter does not support flushing")
+				spanTokens.RecordError(err)
+				spanTokens.SetStatus(codes.Error, err.Error())
+				spanTokens.End()
 				return
 			}
 		}
+
+		// Statisitcs for metrics and traces attributes
+		allTokensElapsedTimeMilisecond := float64(time.Since(allTokensStartTime)) / float64(time.Millisecond)
+		allTokenLatencyMeasure.Record(ctx, allTokensElapsedTimeMilisecond)
+		spanTokens.SetAttributes(attribute.Float64("llm.first.token.latency.ms", firstTokenLatencyMilliseconds))
+		spanTokens.SetAttributes(attribute.Float64("llm.next.token.latency.total.ms", nextTokenLatencyTotal))
+		spanTokens.SetAttributes(attribute.Float64("llm.next.token.latency.count", nextTokenLatencyCount))
+		spanTokens.SetAttributes(attribute.Float64("llm.next.token.latency.avg.ms", nextTokenLatencyTotal/nextTokenLatencyCount))
```

Comment:
Total and count are already reported, so I think this derived value is redundant.